server/lib/actions_controllers.js

```diff
@@ -1,87 +1,68 @@
-const { someMatch } = require('builders/utils')
 const error_ = require('lib/error/error')
 const validateObject = require('lib/validate_object')
 const { rolesByAccess } = require('./user_access_levels')
-const { send } = require('./responses')
-const { sanitize } = require('./sanitize/sanitize')
-const { track } = require('./track')
-const assert_ = require('./utils/assert_types')
+const { controllerWrapper, validateControllerWrapperParams } = require('./controller_wrapper')
+
+// A function to route requests to an endpoint to sub-endpoints
+// identified by their 'action' names:
+//
+// controllers = {
+//   [accessLevel]: {
+//     [actionA]: controller,
+//     [actionB]: {
+//       sanitization,
+//       controller,
+//       track,
+//     },
+//   }
+// }
 
 module.exports = controllers => {
-  const actions = getActions(controllers)
+  const actionsControllersParams = getActionsControllersParams(controllers)
   return async (req, res) => {
     // Accepting the action to be passed either as a query string
     // or as a body parameter for more flexibility
-    const action = req.query.action || req.body.action
+    let action = req.query.action || req.body.action
 
-    if (action == null && actions.default == null) {
+    if (action == null && actionsControllersParams.default == null) {
```

Comment:
strickly(?) equivalent suggestion (that cannot fit inside suggestion commenting tool)

```
    const action = req.query.action || req.body.action || 'default'

    if (action === 'default' && actionsControllersParams.default == null) {
```

and remove `action = action || 'default'` below

Reply:
refactor done in f26e02c67